src/Francis.js

```diff
@@ -119,31 +120,42 @@ class Francis {
             finalize: function () { self.logger.log('yay, it worked :)'); }
         };
 
-        context.transactionId = 'something'; // Need a uuid
+        self.transactions[transactionName].context.transactionId = 'something'; // Need a uuid
     }
 
     _removeTransaction (transactionName){
         let self = this;
-        self.status = Francis.QUERIER_STATUS.CLEANUP;
+        //self.status = Francis.QUERIER_STATUS.CLEANUP; //I don't think it's necessary to change the global status whenever you're removing a transaction
         self._abortTransaction(transactionName);
-        self.transactions = self.transactions.filter(transaction => transaction.name !== transactionName);
-        self.status = Francis.QUERIER_STATUS.READY;
+        //Filter won't work on an object (just checked)
+        //self.transactions = self.transactions.filter(transaction => transaction.name !== transactionName);
+        delete self.transactions[transactionName];
+        //self.status = Francis.QUERIER_STATUS.READY; //Same comment as above
     }
 
     _abortTransaction(transactionName){
         let self = this;
+
         let transaction = self.transactions[transactionName];
+        //If the transaction is already aborting, just return.
+        if(transaction.status === Francis.QUERIER_STATUS.ABORTING) return false;
+
         let transactionId = transaction.context.transactionId;
-        let requestsToAbort = self.workingRequests.filter(request => request.transactionId === transactionId);
 
         //Set up aborting state
         transaction.status = Francis.QUERIER_STATUS.ABORTING;
-        //Abort each request for the transaction
-        requestsToAbort.forEach(request => request.xhr.abort());
-        //remove all requests related to that transaction from the working set
-        self.workingRequests = self.workingRequests.filter(request => request.transactionId !== transactionId);
+
+        //What are your thoughts on this change, should minimize the amount of looping over stuff.
+        for(let i = 0; i < self.workingRequests.length; i++) {
+            let workItem = self.workingRequests[i];
+            if(workItem.transactionId === transactionId) {
+                workItem.xhr.abort();
+                self.workingRequests.splice(i,1);
```

Comment:
You're right, should just keep track of them and then execute splicing at the end.